packages/addons/addon-depends/jre-depends/jre-libXinerama/package.mk

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
+
+. $(get_pkg_directory libXinerama)/package.mk
+
+PKG_NAME="jre-libXinerama"
+PKG_LONGDESC="libXinerama for JRE"
+PKG_URL=""
```

Comment:
Add `"PKG_DEPENDS_UNPACK=libXinerama"` here, otherwise you may not have any source tarball to unpack - see https://github.com/LibreELEC/LibreELEC.tv/blob/027def8c892257115f8d6ebe471226ac2041136b/packages/addons/addon-depends/chrome-depends/chrome-libXcomposite/package.mk as an example.

Reply:
I don't understand how this PKG_DEPENDS_UNPACK works but on LE 9.0 branch doesn't do anything. In case there is no source it doesn't get downloaded either with this option. I looked through sources and I don't see it's call.
The only thing which works is to add get call under unpack() function. I wrote about this issue on IRC.

Reply:
Delete all your `libXinerama` source tarballs then try and build this addon (and just this add-on) - it will fail as you won't have any `libXinerama` source tarball to unpack. You don't have any actual dependency on `libXinerama` so there's no reason for the build system to download the `libXinerama` tarball (as it would if it were building the `libXinerama` package).

`PKG_DEPENDS_UNPACK` is supported in master and libreelec-9.2 and instructs the build system to unpack (and therefore, download) the named packages prior to unpacking this package. In this case you will incur the overhead of unpacking the named packages, but will at least ensure your package always has a tarball to unpack/patch/whatever.

As I say look at the existing examples where this already works, with or without a pre-existing tarball.

Sorry, I don't really follow IRC.

Reply:
Ok, then this is something which doesn't work in libreelec-9.0. Good to know. Will re-add this to all the packages.

Reply:
Sorry, quick correction to my original post, the line to be added should read:
```
PKG_DEPENDS_UNPACK+=" libXinerama"
```
ie. append the package name in case the package being sourced already includes a `PKG_DEPENDS_UNPACK`.

Reply:
Yes, already saw how it's done for chrome depends.

Reply:
> Ok, then this is something which doesn't work in libreelec-9.0.

Yes, this "package inheritance" approach only ever worked in `libreelec-9.0` if you had already downloaded the tarball of the "parent" package (either downloaded manually, or by building the "parent" package weeks/months/years previously and still having the tarball in your repo). Which is to say, it only worked by chance for most people.

Now, chance is no longer part of the equation. :)